server.go

```diff
@@ -880,8 +880,9 @@ func SendHeader(ctx context.Context, md metadata.MD) error {
 }
 
 // SetTrailer sets the trailer metadata that will be sent when an RPC returns.
-// It may be called at most once from a unary RPC handler. The ctx is the RPC
-// handler's Context or one derived from it.
+// It may be called multiple times from a unary RPC handler.
+// When called more than once, all the provided metadata will be merged.
```

Comment:
Until I read the code and found the slice append, I was worried that it would "merge" maps by overwriting existing keys.